docs: Swap phases 4 and 5 in washing machine workflow (#6244)
adobe/spectrum-web-components: CONTRIBUTOR-DOCS/03_project-planning/02_workstreams/02_2nd-gen-component-migration/03_migration-project-planning.md

Reviewer: I was just coming to comment on this list item but realized it appears twice if you're up for removing this one 🙂

Reviewer: I think this should be Ticket 6, with five prior phase PRs:

```suggestion
- **Ticket 6 (Phase 8: Review):** After **all five** prior phase PRs are merged into the feature branch, run **final Q&A** on the full integration (lint, tests, Storybook, checklist, status table—see the [washing machine workflow](02_step-by-step/01_washing-machine-workflow.md) Phase 8). Then **merge the feature branch to `main`** (or your repo’s default integration branch). Phase 8 is the last gate before that merge.
```

Reviewer: These ticket numbers need updating to 1-5 instead of 1-7:

```suggestion
- **Tickets 1–5:** Each ticket is delivered as a **PR merged into that feature branch**. Close the ticket when the PR is merged; the branch should always carry the integrated result of completed phases.
```

| 1 | Migrate [Component] — Preparation | Phase 1: Preparation |
| 2 | Migrate [Component] — Setup | Phase 2: Setup |
| 3 | Migrate [Component] — API migration | Phase 3: API migration |

> | 4 | Migrate [Component] — Accessibility | Phase 4: Accessibility |

Reviewer: Are we combining Accessibility with API migration? Should we reflect that in this tickets table (and the one below)? Or leave a note somewhere about those two phases being combined (as one ticket)?

Author: We are combining it in _Jira_, but I think they should still be two separate phases. I can make a note about this somewhere

Author: I updated the references to Ticket numbers in the file--they should all reference ticket 5 now since that's where the style guide and testing convention step lives now.

Reviewer: Let me know if I'm missing something about the way this works 😬

I read this and take it to mean there are 8 phases and 8 tickets so there's a 1:1 mapping:

<img width="862" height="765" alt="image" src="https://github.com/user-attachments/assets/56d69a85-44b3-4716-8afe-f8284345b40b" />

But are the ticket lists here in GitHub not meant to match what we have in Jira?

<img width="396" height="335" alt="image" src="https://github.com/user-attachments/assets/9bb95be1-1fe7-4f45-ab0c-421ee3215939" />

Author: oHHHH i totally missed this file. I get what you're asking. I'll update this

Author: I've updated the table. The thing that might be confusing is the reference to the a11y doc, which I've included here as "step 0" because even though it's not part of the WM workflow, it still matters in the context of the epic/merging feature branch, right? 

How do you feel about this? I can remove the reference altogether.

> | Preparation | 1 | Migration plan; component analysis read or created; breaking changes listed. |
> | Setup | 2 | Core and SWC folders and files created; build passes; component importable. |
> | API migration | 3 | Types in core; base and SWC API; JSDoc and @internal. |
> | Accessibility | 4 | APG pattern applied; ARIA and keyboard; a11y tests. |
> | Styling | 5 | CSS migrated; tokens and variants; render() with classMap. |
> | Testing | 6 | Unit and a11y tests pass; coverage in place. |
> | Documentation | 7 | JSDoc; Storybook stories; migration notes if needed. |
> | Review & PR | 8 | Final Q&A on feature branch with phases 1–7 merged; checklist and status table done; **feature branch merged to `main`**. |

Reviewer: What do you think about adjusting the ticket column to look something like this to map more clearly to ticket titles in the previous table?

| Ticket | Phase | Key deliverables |
|--------|--------|-------------------|
| Migration plan | 1 | Migration plan; component analysis read or created; breaking changes listed. |
| File structure, API, Typescript, and accessibility | 2 | Core and SWC folders and files created; build passes; component importable. |
| File structure, API, Typescript, and accessibility | 3 | Types in core; base and SWC API; JSDoc and @internal. |
| File structure, API, Typescript, and accessibility | 4 | APG pattern applied; ARIA and keyboard; a11y tests. |
| Full visual fidelity | 5 | CSS migrated; tokens and variants; render() with classMap. |
| Code style conformance and testing | 6 | Unit and a11y tests pass; coverage in place. |
| Documentation and consumer migration guide | 7 | JSDoc; Storybook stories; migration notes if needed. |
| Review & PR | 8 | Final Q&A on feature branch with phases 1–7 merged; checklist and status table done; **feature branch merged to `main`**. |

Here's the suggestion:

```suggestion
| Ticket | Phase | Key deliverables |
|--------|--------|-------------------|
| Migration plan | 1 | Migration plan; component analysis read or created; breaking changes listed. |
| File structure, API, Typescript, and accessibility | 2 | Core and SWC folders and files created; build passes; component importable. |
| File structure, API, Typescript, and accessibility | 3 | Types in core; base and SWC API; JSDoc and @internal. |
| File structure, API, Typescript, and accessibility | 4 | APG pattern applied; ARIA and keyboard; a11y tests. |
| Full visual fidelity | 5 | CSS migrated; tokens and variants; render() with classMap. |
| Code style conformance and testing | 6 | Unit and a11y tests pass; coverage in place. |
| Documentation and consumer migration guide | 7 | JSDoc; Storybook stories; migration notes if needed. |
| Review & PR | 8 | Final Q&A on feature branch with phases 1–7 merged; checklist and status table done; **feature branch merged to `main`**. |
```
